pandas/core/frame.py

```diff
@@ -231,6 +231,21 @@ class DataFrame(NDFrame):
     >>> df3 = DataFrame(np.random.randn(10, 5),
     ...                 columns=['a', 'b', 'c', 'd', 'e'])
 
+    Leaving dtype=None in constructor will infer a wider type than necessary
```

Comment:
can you show an a typical example of passing a dtype, with a single type, then multiple types, then this example.